From 785b8cd4bffd47c4e0cfc9d7d2ec6288738ba231 Mon Sep 17 00:00:00 2001
From: Alison Wang <alison.wang@nxp.com>
Date: Tue, 20 Nov 2018 10:33:01 +0800
Subject: [PATCH 348/706] arm64: dts: ls1028a: Update audio nodes for
 LS1028ARDB board

This patch updates audio nodes for LS1028ARDB board.

Signed-off-by: Alison Wang <alison.wang@nxp.com>
(cherry picked from commit 9104d16b71db7d2e9310a3bce6cbfd7dd841647b)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 .../boot/dts/freescale/fsl-ls1028a-rdb.dts     | 18 +++++++++---------
 arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi | 16 ++++++++--------
 2 files changed, 17 insertions(+), 17 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts b/arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts
index 5b87b0541bb7..aa0796ad990f 100755
--- a/arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts
@@ -64,14 +64,14 @@
 	sys_mclk: clock-mclk {
 		compatible = "fixed-clock";
 		#clock-cells = <0>;
-		clock-frequency = <24576000>;
+		clock-frequency = <25000000>;
 	};
 
-	reg_3p3v: regulator-3p3v {
+	reg_1p8v: regulator-1p8v {
 		compatible = "regulator-fixed";
-		regulator-name = "3P3V";
-		regulator-min-microvolt = <3300000>;
-		regulator-max-microvolt = <3300000>;
+		regulator-name = "1P8V";
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <1800000>;
 		regulator-always-on;
 	};
 
@@ -91,7 +91,7 @@
 			"Speaker Ext", "LINE_OUT";
 
 		simple-audio-card,cpu {
-			sound-dai = <&sai2>;
+			sound-dai = <&sai4>;
 			frame-master;
 			bitclock-master;
 		};
@@ -100,7 +100,7 @@
 			sound-dai = <&codec>;
 			frame-master;
 			bitclock-master;
-			system-clock-frequency = <24576000>;
+			system-clock-frequency = <25000000>;
 		};
 	};
 };
@@ -122,8 +122,8 @@
 				#sound-dai-cells = <0>;
 				compatible = "fsl,sgtl5000";
 				reg = <0xa>;
-				VDDA-supply = <&reg_3p3v>;
-				VDDIO-supply = <&reg_3p3v>;
+				VDDA-supply = <&reg_1p8v>;
+				VDDIO-supply = <&reg_1p8v>;
 				clocks = <&sys_mclk>;
 			};
 		};
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
index f321153cef3a..16215f9ff9d8 100755
--- a/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
@@ -674,8 +674,8 @@
 			dma-channels = <32>;
 			big-endian;
 			clock-names = "dmamux0", "dmamux1";
-			clocks = <&clockgen 4 3>,
-				 <&clockgen 4 3>;
+			clocks = <&clockgen 4 1>,
+				 <&clockgen 4 1>;
 		};
 
 		sai1: sai@f100000 {
@@ -683,8 +683,8 @@
 			compatible = "fsl,vf610-sai";
 			reg = <0x0 0xf100000 0x0 0x10000>;
 			interrupts = <0 82 0x4>;
-			clocks = <&clockgen 4 3>, <&clockgen 4 3>,
-				 <&clockgen 4 3>, <&clockgen 4 3>;
+			clocks = <&clockgen 4 1>, <&clockgen 4 1>,
+				 <&clockgen 4 1>, <&clockgen 4 1>;
 			clock-names = "bus", "mclk1", "mclk2", "mclk3";
 			dma-names = "tx", "rx";
 			dmas = <&edma0 1 4>,
@@ -697,8 +697,8 @@
 			compatible = "fsl,vf610-sai";
 			reg = <0x0 0xf110000 0x0 0x10000>;
 			interrupts = <0 82 0x4>;
-			clocks = <&clockgen 4 3>, <&clockgen 4 3>,
-				 <&clockgen 4 3>, <&clockgen 4 3>;
+			clocks = <&clockgen 4 1>, <&clockgen 4 1>,
+				 <&clockgen 4 1>, <&clockgen 4 1>;
 			clock-names = "bus", "mclk1", "mclk2", "mclk3";
 			dma-names = "tx", "rx";
 			dmas = <&edma0 1 6>,
@@ -711,8 +711,8 @@
 			compatible = "fsl,vf610-sai";
 			reg = <0x0 0xf130000 0x0 0x10000>;
 			interrupts = <0 83 0x4>;
-			clocks = <&clockgen 4 3>, <&clockgen 4 3>,
-				 <&clockgen 4 3>, <&clockgen 4 3>;
+			clocks = <&clockgen 4 1>, <&clockgen 4 1>,
+				 <&clockgen 4 1>, <&clockgen 4 1>;
 			clock-names = "bus", "mclk1", "mclk2", "mclk3";
 			dma-names = "tx", "rx";
 			dmas = <&edma0 1 10>,
-- 
2.17.1

